### ISSUE TITLE
[Feat Request] Add documentation for TIN Algorithm

### ISSUE THREAD

**Issue by ar-siddiqui** [qgis/QGIS-Documentation#7792]
## Description

Page URL: https://docs.qgis.org/testing/en/docs/user_manual/processing_algs/qgis/interpolation.html#qgistininterpolation

I am trying to use the TIN Interpolation algorithm and realized there is no documentation describing the types [Points, Break Lines, Structure Lines], specifically, I couldn't understand the difference between structures and break lines and how the tool utilizes them internally. It would be nice to have a description of that in the documentation and tool help.

![image](https://user-images.githubusercontent.com/53625184/190697721-b64df522-9715-439e-9c51-059dab2a440e.png)

![image](https://user-images.githubusercontent.com/53625184/190697738-57b4c06b-6ccd-4e67-bb3f-7bb20771ab41.png)

**Comment by DelazJ** [comment]
@ar-siddiqui you are right, indeed. And this would also apply to [IDW Interpolation](https://docs.qgis.org/testing/en/docs/user_manual/processing_algs/qgis/interpolation.html#qgisidwinterpolation)
[This description](https://github.com/qgis/QGIS/blob/d326d38660734d48e48e0d4baf9cd4c084626bd3/src/analysis/interpolation/qgsinterpolator.h#L81-L83) is so far the most verbose description I could find in QGIS code repo. Browsing [2.18 docs](https://docs.qgis.org/2.18/en/docs/user_manual/plugins/plugins_interpolation.html#interpolation-plugin), I also found
>Note that using lines as constraints for the interpolation the triangulation (TIN method) you can either use ‘structure lines’ or ‘break lines’. When using ‘break lines’ you produce sharp breaks in the surface while using ‘structure lines’ you produce continuous breaks. The triangulation is modified by both methods such that no edge crosses a breakline or structure line.

 We definitely would need help of a developer or someone using these features for sufficient explanation.
